### PR TITLE
Checking of private_generics in the testing framework.

### PR DESCRIPTION
**Pull request by steelgeek091**
The functionality to check `private_generics` and other function attributes has been implemented in `rooch-integration-test-runner` and `rooch-framework-tests`.

The example test case is in the file `crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.move`. 

The current check result of the test is `incorrect`, so the following content has been written in the exp file, which is the content of the file `crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.exp`:

```txt
processed 4 tasks

task 1 'publish'. lines 3-8:
status EXECUTED

task 2 'publish'. lines 10-48:
error: resource type "KeyStruct" in function "0x2::object_storage::remove" not defined in current module or not allowed
   ┌─ /tmp/tempfile:37:19
   │
37 │         let obj = object_storage::remove<KeyStruct>(object_storage, object_id);
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: resource type "KeyStruct" in function "0x2::object::unpack" not defined in current module or not allowed
   ┌─ /tmp/tempfile:39:34
   │
39 │         let (_id,_owner,value) = object::unpack(obj);
   │                                  ^^^^^^^^^^^^^^^^^^^

error: resource type "KeyStruct" in function "0x2::account_storage::global_move_to" not defined in current module or not allowed
   ┌─ /tmp/tempfile:40:9
   │
40 │         account_storage::global_move_to(ctx, &sender, value);
   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

status EXECUTED

task 3 'run'. lines 50-57:
status EXECUTED
```

Additionally, some test cases in the directory `crates/rooch-framework-tests/tests/cases/entry_function` need to be rewritten. @WGB5445

### CODE REVIEW

File: crates/rooch-integration-test-runner/src/lib.rs [hunk 1]
```diff
@@ -68,10 +69,10 @@ impl<'a> MoveTestAdapter<'a> for MoveOSTestAdapter<'a> {
 
     fn init(
         default_syntax: SyntaxChoice,
-        pre_compiled_deps: Option<&'a FullyCompiledProgram>,
-        task_opt: Option<TaskInput<(InitCommand, Self::ExtraInitArgs)>>,
+        option: Option<&'a FullyCompiledProgram>,
```
Comment (jolestar) [inline crates/rooch-integration-test-runner/src/lib.rs:72]: `option` is not a good name. How about keeping the `pre_compiled_deps`?

File: crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.move [hunk 2]
```diff
@@ -1,29 +1,57 @@
...
         x: u64,
     }
+    use std::string::{Self};
 
-    #[private(T1)]
-    public fun publish_foo<T1>(ctx: &mut StorageContext, s: &signer) {
-        account_storage::global_move_to<Foo>(ctx, s, Foo { x: 500 })
+    #[private_generics(T1)]
+    public fun publish_foo<T1>(s: &signer) {
+        move_to<Foo>(s, Foo { x: 500 })
```
Comment (jolestar) [inline crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.move:19]: Do not use `move_to`. The Move global instructions will be deprecated in #248

File: moveos/moveos/src/moveos_test_runner.rs [hunk 3]
```diff
@@ -0,0 +1,964 @@
...
+    deps: impl Iterator<Item = &'a CompiledModule>,
+    file_name: &str,
+) -> Result<CompiledScript> {
+    use move_ir_compiler::Compiler as IRCompiler;
+    let code = std::fs::read_to_string(file_name).unwrap();
+    let (script, _) = IRCompiler::new(deps.collect()).into_compiled_script_and_source_map(&code)?;
+    Ok(script)
+}
+
+pub trait MoveOSMainTestAdapter<'a>: Sized {
```
Comment (jolestar) [inline moveos/moveos/src/moveos_test_runner.rs:332]: How about naming it `MoveOSTestAdapter`?

File: crates/rooch-integration-test-runner/src/lib.rs (hunk 1 above)
Comment (steelgeek091) [reply crates/rooch-integration-test-runner/src/lib.rs:72]: Alright, it's possible that I made a mistake and overlooked it.

File: crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.move (hunk 2 above)
Comment (steelgeek091) [reply crates/rooch-integration-test-runner/tests/cases/private_generics_indirect_call.move:19]: Yes, the `move_to` has been replaced with `account_storage::global_move_to`.

File: moveos/moveos/src/moveos_test_runner.rs (hunk 3 above)
Comment (steelgeek091) [reply moveos/moveos/src/moveos_test_runner.rs]: Okay, the name has been changed.

**Pull request merged**